sklearn/utils/validation.py

```diff
@@ -1655,6 +1655,51 @@ def check_is_fitted(estimator, attributes=None, *, msg=None, all_or_any=all):
         raise NotFittedError(msg % {"name": type(estimator).__name__})
 
 
+def _estimator_has(attr, delegates=None):
+    """Check if we can delegate a method to the underlying estimator.
+
+    We check the `delegates` in the order they are passed.
+    By default, we first check the fitted estimator if available,
+    otherwise we check the unfitted estimator.
```

Comment:
nitpicks to use the full line length

```suggestion
    We check the `delegates` in the order they are passed. By default, we first check
    the fitted estimator if available, otherwise we check the unfitted estimator.
```